lib/Extractor/Candidates.cpp

```diff
@@ -492,6 +497,11 @@ Inst *ExprBuilder::get(Value *V) {
   if (!E) {
     E = build(V);
   }
+  E->DemandedBitsVal = APInt(E->Width, 0, false);
+  if (HarvestDemandedBits)
+    if (Instruction *I = dyn_cast<Instruction>(V))
+      if (DB && (DB->getDemandedBits(I)).getBoolValue())
```

Comment:
can you explain the purpose of checking DB against null?
also, what is the purpose of calling getDemandedBits() twice, the first time only to check if there are any bits?